pydra/tasks/core/sequences.py

```diff
@@ -0,0 +1,117 @@
+import attr
+import typing as ty
+from pydra.engine.specs import BaseSpec, SpecInfo
+from pydra.engine.core import TaskBase
+from pydra.engine.helpers import ensure_list
+
+
+@attr.s(kw_only=True)
+class MergeInputSpec(BaseSpec):
+    axis: ty.Literal["vstack", "hstack"] = attr.ib(
+        default="vstack",
+        metadata={
+            "help_string": "Direction in which to merge, hstack requires same number of elements in each input."
+        },
+    )
+    no_flatten: bool = attr.ib(
+        default=False,
+        metadata={
+            "help_string": "Append to outlist instead of extending in vstack mode."
+        },
+    )
+    ravel_inputs: bool = attr.ib(
+        default=False,
+        metadata={"help_string": "Ravel inputs when no_flatten is False."},
+    )
+
+
+def _ravel(in_val):
+    if not isinstance(in_val, list):
+        return in_val
+    flat_list = []
+    for val in in_val:
+        raveled_val = _ravel(val)
+        if isinstance(raveled_val, list):
+            flat_list.extend(raveled_val)
+        else:
+            flat_list.append(raveled_val)
+    return flat_list
+
+
+class Merge(TaskBase):
+    """
+    Task to merge inputs into a single list
+
+    ``Merge(1)`` will merge a list of lists
+
+    Examples
+    --------
+    >>> from pydra.tasks.core.sequences import Merge
+    >>> mi = Merge(3, name="mi")
+    >>> mi.inputs.in1 = 1
+    >>> mi.inputs.in2 = [2, 5]
+    >>> mi.inputs.in3 = 3
+    >>> out = mi()
+    >>> out.output.out
+    [1, 2, 5, 3]
+
+    >>> merge = Merge(1, name="merge")
+    >>> merge.inputs.in1 = [1, [2, 5], 3]
+    >>> out = merge()
+    >>> out.output.out
+    [1, [2, 5], 3]
+
+    >>> merge = Merge(1, name="merge")
+    >>> merge.inputs.in1 = [1, [2, 5], 3]
+    >>> merge.inputs.ravel_inputs = True
+    >>> out = merge()
+    >>> out.output.out
+    [1, 2, 5, 3]
+
+    >>> merge = Merge(1, name="merge")
+    >>> merge.inputs.in1 = [1, [2, 5], 3]
+    >>> merge.inputs.no_flatten = True
+    >>> out = merge()
+    >>> out.output.out
+    [[1, [2, 5], 3]]
+    """
+
+    _task_version = "1"
+
+    def __init__(self, numinputs, *args, **kwargs):
+        self._numinputs = max(numinputs, 0)
+        self.input_spec = SpecInfo(
+            name="Inputs",
+            fields=[(f"in{i + 1}", ty.List) for i in range(self._numinputs)],
+            bases=(MergeInputSpec,),
+        )
+        # Can't set as class attribute
+        self.output_spec = SpecInfo(
+            name="Outputs", fields=[("out", ty.List)], bases=(BaseSpec,)
+        )
```

Comment:
I was surprised by this constraint. Setting it as a class attribute caused this to fail with a `KeyError`:

https://github.com/nipype/pydra/blob/b934e4b549fa4e16cdffff844dcf655cdf0a4c13/pydra/engine/core.py#L199-L202